etna/transforms/timestamp/event.py

```diff
@@ -0,0 +1,155 @@
+from typing import List
+from typing import Optional
+from typing import Dict
+import numpy as np
+import pandas as pd
+
+from etna.datasets import TSDataset
+from etna.transforms.base import IrreversibleTransform
+from etna.distributions import BaseDistribution, IntDistribution, CategoricalDistribution
+
+
+class EventTransform(IrreversibleTransform):
+    """EventTransform marks days before and after event
+
+    In ``binary`` mode shows whether there will be or were events regarding current date.
+
+    In ``distance`` mode shows distance to the previous and future events regarding current date.
+    Computed as :math:`1 / x`, where x is a distance to the nearest event.
+    """
+
+    def __init__(
+        self,
+        in_column: str,
+        out_column: str,
+        n_pre: int = 1,
+        n_post: int = 1,
+        mode: str = 'binary'
+    ):
+        """
+        Init EventTransform.
+
+        Parameters
+        ----------
+        in_column:
+            binary column with event indicator.
+        out_column:
+            base for creating out columns names.
+        n_pre:
+            number of days before the event to react.
+        n_post:
+            number of days after the event to react.
+        mode: {'binary', 'distance'}, default='binary'
+            Specify mode of marking events:
+
+            - `'binary'`: whether there will be or were events regarding current date in binary type;
+            - `'distance'`: distance to the previous and future events regarding current date;
+
+        Raises
+        ------
+        TypeError:
+            Type of `n_pre` or `n_post` is different from `int`.
+        TypeError:
+            Value of `mode` is not in ['binary', 'distance'].
+        """
+        if not isinstance(n_pre, int) or not isinstance(n_post, int):
```

Comment:
You don't need to check it explicitly, we add typing for such purpose -- in the perfect world each argument in each constructor should be validated like this but for this purpose better to use sth like [pydantic](https://github.com/pydantic/pydantic)

Reply:
We should do it everywhere or don't do it at all